homeassistant/components/stream/hls.py

```diff
@@ -109,14 +137,67 @@ def name(self) -> str:
     @property
     def format(self) -> str:
         """Return container format."""
-        return "mpegts"
+        return "mp4"
```

Comment:
Can we not use `fmp4` to create the fragmented MP4 directly instead of during HLS rendering or is that not an option in FFMPEG out of the box?

Reply:
> 
> 
> In general this looks good. What does desktop browser support look like? I left a few comments comparing to the [hlsenc](https://github.com/FFmpeg/FFmpeg/blob/master/libavformat/hlsenc.c) provided by FFMPEG. Searching that file for `SEGMENT_TYPE_FMP4` shows the additional logic they have to support `m4s`. Obviously we aren't using that as it writes files to disk, but perhaps it's a good reference for our (simplified) implementation.

Yes, the reason I used this approach is because PyAV isn't written to deal with multiple file IO in ffmpeg.
I had to do a bit of digging into mp4 container format specifications and figure out the differences between mp4 and fmp4.
mp4 containers are set up with a hierarchy of boxes, and while most of the data is the same, there is some slight editing needed to convert regular mp4 to fmp4.
I discussed browser support in the forum post here: https://community.home-assistant.io/t/add-support-for-h-265-in-stream-component/205661 . Basically, Windows you are SOL unless you use a custom chromium or if/when Microsoft decides to add H.265 functionality back into the new Edge.
Here are a few links on mp4/fmp4 for your reference:
https://stackoverflow.com/questions/35177797/what-exactly-is-fragmented-mp4fmp4-how-is-it-different-from-normal-mp4
https://www.ramugedia.com/mp4-container
https://www.cimarronsystems.com/wp-content/uploads/2017/04/Elements-of-the-H.264-VideoAAC-Audio-MP4-Movie-v2_0.pdf
https://observablehq.com/@benjamintoofer/iso-base-media-file-format
https://www.programmersought.com/article/99832446840/
https://www.file-recovery.com/mp4-signature-format.htm